### ISSUE TITLE
High Memory Usage (472 megabytes)

### ISSUE THREAD

**Issue by tgharib** [syncany/syncany#382]
Daemon: 284 megabytes
GUI: 188 megabytes

I know you guys don't have much control over memory usage since you are using Java but maybe some optimizations could be done in the long-term future? In comparison, the MEGA client uses less than 30 megabytes of RAM.

I am running Manjaro Linux XFCE 64-bit edition for anyone wondering.

**Comment by binwiederhier** [comment]
Those are some really small numbers :) I have seen Syncany consume >3GB of memory. And that's not terrible, because we're doing *a lot* (let me emphasize that even more: **a lot**) more magic on the client side. Deduplication takes resources. Period. 

However, as part of #195, we will likely tackle this as well at least to some extent. I'd be happy to reach a memory consumption of 500 MB for 200k file syncing. That'd be awesome. 

Right now, I don't have a feel for what is high and what is low. I haven't run a profiler even once, so I don't know where the memory is coming from -- which is a good thing, because there are likely many, many things that can be optimized. 

Thanks for reporting this, I'll add it to the #195 subtask list.

**Comment by binwiederhier** [comment]
@tgharib If you have experience with Java Profilers (or if you'd be willing to research/test/experiment with it), I'd love that and everyone would benefit from your input. We certainly haven't done it yet. I know that VisualVM and jps can do all sorts of things, but there are certainly other tools out there. :+1:

**Comment by binwiederhier** [comment]
A bit of research and VisualVM tell me that the used heap size is a lot smaller than the allocated heap size. Also: Just calling `java -version` will initialize the heap with 260 MB (on my system), so it's fair to say that Java requires a lot of resources to start with.

To play with:
- `-Xms`, `-Xmx` --> http://stackoverflow.com/a/6452812/1440785 
- `<maxMemory>` in `daemon.xml` --> http://syncany.readthedocs.org/en/latest/configuration.html?highlight=maxmemory#options-for-userconfig-xml
- http://www.mkyong.com/java/find-out-your-java-heap-memory-size/

Also (no idea what to make of that):

![2015021523 04 29_java visualvm_001](https://cloud.githubusercontent.com/assets/664597/6207145/b9bf35c0-b568-11e4-86a3-446779583797.png)
![2015021523 05 17_java visualvm_001](https://cloud.githubusercontent.com/assets/664597/6207147/bc902dea-b568-11e4-9839-61e970834ee0.png)

**Comment by binwiederhier** [comment]
Adjusting `-Xms` (initial heap size) seems to do the trick. `-Xms30m` looks like this. However, Java will always adjust the size up until `-Xmx` is reached. Default if 2 GB. Maybe we should let users adjust `-Xms` and `-X...` manually.

![2015021523 21 04_java visualvm_001](https://cloud.githubusercontent.com/assets/664597/6207178/8b78d134-b569-11e4-9ef4-624bc5ec0b86.png)

**Comment by pimotte** [comment]
I have good hopes that #364 will magically resolve this. (At least to keep the memory usage sub 200MB during all usecases.)

Of all possible optimizations, memory usage is at the top of the list for me. That said, it still is an optimization, so it has relatively low priority right now.

**Comment by tgharib** [comment]
How did you pass the -Xms30m flag? I tried sy gui -Xms30m but that didn't seem to affect memory usage.

I have /usr/bin/sy symlinked to /usr/bin/syncany. /usr/bin/syncany seems to be a shell script that launches the application.

**Comment by pimotte** [comment]
We support setting maximum memory through a config file. You can set -Xms30m by "hacking" it into /usr/bin/syncany. There is a proper way to do that, but I can't find the source material for this script for the life of me.

Memory usage in the GUIs likely due to the fact that if you start a GUI, while no daemon is running, it starts the daemon in the same JVM.

**Comment by binwiederhier** [comment]
The script is generated by Gradle, we change things in it via [application.distribution.gradle](https://github.com/syncany/syncany/blob/develop/gradle/gradle/application.distribution.gradle#L106). 

@pimotte What do you think? Should we offer a more generic way to set JVM parameters? 

(a) Like this:
```xml
<jvm>
  <Xms>100m</Xms>
  <Xmx>1024m</Xmx>
</jvm>
```

(b) Or maybe just:
```xml
<jvm>-Xms100m -Xmx1024m</jvm>
```

(c) Or more user friendly:
```xml
<maxMemory>1024m</maxMemory> <!-- exists! -->
<initialMemory>100m</initialMemory>
```

The latter way is obviously not generic at all. We'd need to pick and map which params we support.

**Comment by cr0** [comment]
You're always telling us to protect the user's from themselves. So I strongly disagree with allowing them to mess with JVM settings. *We* should find the best settings for them and that's it!

**Comment by pimotte** [comment]
The best settings highly depend on whether Syncany is run on a Raspberry Pi or Philipp's laptop. (At least for the maxMemory.)

Is there any reason not to set the initialMemory to some negligent amount (5mB)?

If this is not a good option for one reason or another, I have no problem making this configurable in the .xml. That is protecting the users from themselves, in the sense that if you know enough to edit an .xml file, you should also know enough about RAM to know when your settings will blow up.

**Comment by binwiederhier** [comment]
I'm not sure where we are gathering information regarding the high memory/CPU usage (see #425), but it also just happened on my machine and it was most unpleasant. Here are a few facts and VisualVM screenshots (not very helpful though):

- All 4 CPUs were running at 100% (most unusual, since Java usually just uses one, right?)
- Logging stopped completely (at a random point); weird!!!
- The SWT threads were present, but no icon was visible (and no Python thread was running either
- Many of the XNIO (Undertow worked) threads were 'running'; 
- Thread-8 / -10 / -12 were also running, all of them belonging to `sun.nio.fs.LinuxWatchService.poll`
- Attaching the profiler in VisualVM did not work (it works when the system is not using 100%)
- Versions:

```
Syncany v0.4.5-alpha+SNAPSHOT.1503302150.git89065ba
Syncany Flickr Plugin v0.4.1-alpha
Syncany FTP Plugin v0.4.0-alpha
Syncany GUI Plugin v0.4.5-alpha+SNAPSHOT.1503302146.gitf551120
Syncany Samba Plugin v0.4.1-alpha
Syncany SFTP Plugin v0.4.5-alpha+SNAPSHOT.1503141808.gitb0911cd
Syncany WebDAV Plugin v0.4.0-alpha
```

![visualvm 1 3 3_140](https://cloud.githubusercontent.com/assets/664597/7119961/a96125e2-e201-11e4-861e-fe10a440c6b2.png)
![visualvm 1 3 3_141](https://cloud.githubusercontent.com/assets/664597/7119971/ac282dac-e201-11e4-9bd9-1e21b6a135bb.png)
![visualvm 1 3 3_143](https://cloud.githubusercontent.com/assets/664597/7119974/ae8c6d74-e201-11e4-878d-4cb982bddc26.png)
![visualvm 1 3 3_144](https://cloud.githubusercontent.com/assets/664597/7119975/b03dadea-e201-11e4-926d-1b5f8d9e52f4.png)
![visualvm 1 3 3_145](https://cloud.githubusercontent.com/assets/664597/7119977/b2b77790-e201-11e4-9274-9e36fe3e66b6.png)
![visualvm 1 3 3_146](https://cloud.githubusercontent.com/assets/664597/7119980/b543cac2-e201-11e4-8fa8-6299d116660d.png)
![visualvm 1 3 3_147](https://cloud.githubusercontent.com/assets/664597/7119981/b6b7ffb8-e201-11e4-8985-14639b8aa49c.png)

Thread dump:
```

2015-04-13 17:02:58
Full thread dump OpenJDK 64-Bit Server VM (24.75-b04 mixed mode):

"RMI TCP Connection(16)-127.0.0.1" daemon prio=10 tid=0x00007fd9641d6800 nid=0x1c1e runnable [0x00007fd9447b1000]
   java.lang.Thread.State: RUNNABLE
	at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:566)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.access$400(TCPTransport.java:619)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:684)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:681)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:681)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- <0x00000000f7d11168> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"RMI TCP Connection(15)-127.0.0.1" daemon prio=10 tid=0x00007fd9641fa000 nid=0x1c1d runnable [0x00007fd9462eb000]
   java.lang.Thread.State: RUNNABLE
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.read(SocketInputStream.java:152)
	at java.net.SocketInputStream.read(SocketInputStream.java:122)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:254)
	- locked <0x00000000f7d0ce70> (a java.io.BufferedInputStream)
	at java.io.FilterInputStream.read(FilterInputStream.java:83)
	at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:549)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.access$400(TCPTransport.java:619)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:684)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:681)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:681)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- <0x00000000f7d0caf8> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"*** Profiler Agent Special Execution Thread 6" daemon prio=10 tid=0x00007fd9a4118800 nid=0x1bf4 in Object.wait() [0x00007fd9458e2000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:503)
	at org.netbeans.lib.profiler.server.ProfilerServer$SeparateCmdExecutionThread.run(ProfilerServer.java:289)
	- locked <0x00000000f7cdd4d0> (a java.lang.Object)

   Locked ownable synchronizers:
	- None

"*** JFluid Monitor thread ***" daemon prio=10 tid=0x00007fd9a4638000 nid=0x1bf3 runnable [0x00007fd93b3b2000]
   java.lang.Thread.State: RUNNABLE
	at org.netbeans.lib.profiler.server.Monitors$SurvGenAndThreadsMonitor.updateSurvGenData(Monitors.java:233)
	at org.netbeans.lib.profiler.server.Monitors$SurvGenAndThreadsMonitor.run(Monitors.java:172)

   Locked ownable synchronizers:
	- None

"RMI TCP Connection(17)-127.0.0.1" daemon prio=10 tid=0x00007fd9641db800 nid=0x1bed runnable [0x00007fd98ddc9000]
   java.lang.Thread.State: RUNNABLE
	at java.lang.System.identityHashCode(Native Method)
	at java.io.ObjectOutputStream$HandleTable.hash(ObjectOutputStream.java:2359)
	at java.io.ObjectOutputStream$HandleTable.lookup(ObjectOutputStream.java:2292)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1115)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347)
	at java.util.TreeMap.writeObject(TreeMap.java:2274)
	at sun.reflect.GeneratedMethodAccessor107.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:988)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1495)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1431)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1177)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1547)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1508)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1431)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1177)
	at java.io.ObjectOutputStream.writeArray(ObjectOutputStream.java:1377)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1173)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347)
	at sun.rmi.server.UnicastRef.marshalValue(UnicastRef.java:293)
	at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:332)
	at sun.rmi.transport.Transport$2.run(Transport.java:202)
	at sun.rmi.transport.Transport$2.run(Transport.java:199)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.Transport.serviceCall(Transport.java:198)
	at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:567)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.access$400(TCPTransport.java:619)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:684)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:681)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:681)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- <0x00000000f7cdb870> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"RMI TCP Connection(18)-127.0.0.1" daemon prio=10 tid=0x00007fd964117000 nid=0x1b7c runnable [0x00007fd945de5000]
   java.lang.Thread.State: RUNNABLE
	at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1596)
	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1517)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:1771)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1350)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:370)
	at sun.rmi.server.UnicastRef.unmarshalValue(UnicastRef.java:325)
	at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:307)
	at sun.rmi.transport.Transport$2.run(Transport.java:202)
	at sun.rmi.transport.Transport$2.run(Transport.java:199)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.Transport.serviceCall(Transport.java:198)
	at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:567)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.access$400(TCPTransport.java:619)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:684)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:681)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:681)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- <0x00000000f7cd2f60> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"JMX server connection timeout 11810" daemon prio=10 tid=0x00007fd96002b800 nid=0x1aea runnable [0x00007fd947af9000]
   java.lang.Thread.State: RUNNABLE
	at com.sun.jmx.remote.internal.ServerCommunicatorAdmin.logtime(ServerCommunicatorAdmin.java:210)
	at com.sun.jmx.remote.internal.ServerCommunicatorAdmin.access$400(ServerCommunicatorAdmin.java:32)
	at com.sun.jmx.remote.internal.ServerCommunicatorAdmin$Timeout.run(ServerCommunicatorAdmin.java:158)
	- locked <0x00000000f7cc00e0> (a [I)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"RMI Scheduler(0)" daemon prio=10 tid=0x00007fd9601ee000 nid=0x1ae8 waiting on condition [0x00007fd9442ad000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000f7cb0ee0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:226)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2082)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1090)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:807)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1068)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"RMI TCP Accept-0" daemon prio=10 tid=0x00007fd97c004000 nid=0x1ae2 runnable [0x00007fd9461eb000]
   java.lang.Thread.State: RUNNABLE
	at java.net.PlainSocketImpl.socketAccept(Native Method)
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:398)
	at java.net.ServerSocket.implAccept(ServerSocket.java:530)
	at java.net.ServerSocket.accept(ServerSocket.java:498)
	at sun.management.jmxremote.LocalRMIServerSocketFactory$1.accept(LocalRMIServerSocketFactory.java:52)
	at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.executeAcceptLoop(TCPTransport.java:399)
	at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.run(TCPTransport.java:371)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"Attach Listener" daemon prio=10 tid=0x00007fd9a0002800 nid=0x1901 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"XNIO-1 task-3" prio=10 tid=0x00007fd97801d000 nid=0xd1a waiting on condition [0x00007fd944bb6000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000e0ccc5c8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1068)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"XNIO-3 task-1" prio=10 tid=0x00007fd94c00f000 nid=0xd18 waiting on condition [0x00007fd944db8000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000e0a1cac0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1068)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"XNIO-4 task-1" prio=10 tid=0x00007fd9ac004000 nid=0xd17 waiting on condition [0x00007fd944eb9000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000e0c75a38> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1068)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"XNIO-4 I/O-2" prio=10 tid=0x00007fd964165800 nid=0xd16 runnable [0x00007fd9451db000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0c8a550> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0c8a540> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0c8a428> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-4 I/O-1" prio=10 tid=0x00007fd964164800 nid=0xd15 runnable [0x00007fd9452dc000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0cf5218> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0cf5208> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0c994f0> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"Watcher/Syncany Team" prio=10 tid=0x00007fd970006800 nid=0xd03 waiting on condition [0x00007fd945ae4000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000e0ce4e70> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
	at java.util.concurrent.LinkedBlockingDeque.takeFirst(LinkedBlockingDeque.java:489)
	at java.util.concurrent.LinkedBlockingDeque.take(LinkedBlockingDeque.java:678)
	at sun.nio.fs.AbstractWatchService.take(AbstractWatchService.java:118)
	at org.syncany.operations.watch.DefaultRecursiveWatcher.pollEvents(DefaultRecursiveWatcher.java:80)
	at org.syncany.operations.watch.RecursiveWatcher$1.run(RecursiveWatcher.java:110)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"Thread-12" daemon prio=10 tid=0x00007fd970005000 nid=0xd02 runnable [0x00007fd945be5000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.fs.LinuxWatchService.poll(Native Method)
	at sun.nio.fs.LinuxWatchService.access$600(LinuxWatchService.java:47)
	at sun.nio.fs.LinuxWatchService$Poller.run(LinuxWatchService.java:311)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"Watcher/Office" prio=10 tid=0x00007fd96411c000 nid=0xcde waiting on condition [0x00007fd945fe9000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000e09f2f68> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
	at java.util.concurrent.LinkedBlockingDeque.takeFirst(LinkedBlockingDeque.java:489)
	at java.util.concurrent.LinkedBlockingDeque.take(LinkedBlockingDeque.java:678)
	at sun.nio.fs.AbstractWatchService.take(AbstractWatchService.java:118)
	at org.syncany.operations.watch.DefaultRecursiveWatcher.pollEvents(DefaultRecursiveWatcher.java:80)
	at org.syncany.operations.watch.RecursiveWatcher$1.run(RecursiveWatcher.java:110)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"Thread-10" daemon prio=10 tid=0x00007fd96411a000 nid=0xcdd runnable [0x00007fd9460ea000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.fs.LinuxWatchService.poll(Native Method)
	at sun.nio.fs.LinuxWatchService.access$600(LinuxWatchService.java:47)
	at sun.nio.fs.LinuxWatchService$Poller.run(LinuxWatchService.java:311)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"Watcher/Documents" prio=10 tid=0x00007fd948012800 nid=0xcd9 waiting on condition [0x00007fd9474ee000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000e0a1b810> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
	at java.util.concurrent.LinkedBlockingDeque.takeFirst(LinkedBlockingDeque.java:489)
	at java.util.concurrent.LinkedBlockingDeque.take(LinkedBlockingDeque.java:678)
	at sun.nio.fs.AbstractWatchService.take(AbstractWatchService.java:118)
	at org.syncany.operations.watch.DefaultRecursiveWatcher.pollEvents(DefaultRecursiveWatcher.java:80)
	at org.syncany.operations.watch.RecursiveWatcher$1.run(RecursiveWatcher.java:110)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"Thread-8" daemon prio=10 tid=0x00007fd948011800 nid=0xcd8 runnable [0x00007fd9475ef000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.fs.LinuxWatchService.poll(Native Method)
	at sun.nio.fs.LinuxWatchService.access$600(LinuxWatchService.java:47)
	at sun.nio.fs.LinuxWatchService$Poller.run(LinuxWatchService.java:311)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"HSQLDB Timer @499c8312" daemon prio=10 tid=0x00007fd9601d2800 nid=0xcd5 in Object.wait() [0x00007fd947cfb000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at org.hsqldb.lib.HsqlTimer$TaskQueue.park(Unknown Source)
	- locked <0x00000000e0de4ad0> (a org.hsqldb.lib.HsqlTimer$TaskQueue)
	at org.hsqldb.lib.HsqlTimer.nextTask(Unknown Source)
	- locked <0x00000000e0de4ad0> (a org.hsqldb.lib.HsqlTimer$TaskQueue)
	at org.hsqldb.lib.HsqlTimer$TaskRunner.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"XNIO-3 I/O-4" prio=10 tid=0x00007fd9600a3000 nid=0xcd4 runnable [0x00007fd947dfc000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0a1c6f0> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0a1c700> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0a1c6a8> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-3 I/O-3" prio=10 tid=0x00007fd9600a1800 nid=0xcd3 runnable [0x00007fd947efd000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0a1ff88> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0a1ff78> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0a1cea0> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-3 I/O-2" prio=10 tid=0x00007fd9600a0000 nid=0xcd2 runnable [0x00007fd947ffe000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0a1d440> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0a1d430> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0a1d298> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-3 I/O-1" prio=10 tid=0x00007fd96009f000 nid=0xcd1 runnable [0x00007fd98c18e000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0a1da90> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0a1da80> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0a1d8e8> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-1 task-2" prio=10 tid=0x00007fd98404c000 nid=0xccc waiting on condition [0x00007fd98c28f000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000e0ccc5c8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1068)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"XNIO-2 task-1" prio=10 tid=0x00007fd95c002800 nid=0xccb waiting on condition [0x00007fd98c390000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000e0a27498> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1068)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"XNIO-2 I/O-2" prio=10 tid=0x00007fd964106000 nid=0xcca runnable [0x00007fd98c491000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0a27b28> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0a27b18> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0a27a00> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-2 I/O-1" prio=10 tid=0x00007fd964105000 nid=0xcc9 runnable [0x00007fd98c592000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0a29268> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0a29258> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0a29030> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-1 task-1" prio=10 tid=0x00007fd984043000 nid=0xcc6 waiting on condition [0x00007fd98dbc9000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000e0ccc5c8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2043)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1068)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"GuiWsClient" prio=10 tid=0x00007fd9cc7f4800 nid=0xcbd waiting on condition [0x00007fd98dcca000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at java.lang.Thread.sleep(Native Method)
	at org.syncany.operations.gui.GuiWebSocketClient.connectAndWait(GuiWebSocketClient.java:145)
	at org.syncany.operations.gui.GuiWebSocketClient$1.run(GuiWebSocketClient.java:101)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"XNIO-1 I/O-4" prio=10 tid=0x00007fd9cc767000 nid=0xcb6 runnable [0x00007fd98e0ce000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0cd7af0> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0cd7ae0> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0cd5f20> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-1 I/O-3" prio=10 tid=0x00007fd9cc730000 nid=0xcb5 runnable [0x00007fd98e1cf000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0cd6508> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0cd64f8> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0cd6360> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-1 I/O-2" prio=10 tid=0x00007fd9cc72e800 nid=0xcb4 runnable [0x00007fd98e2d0000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0cdbf00> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0cdbef0> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0cd7ed8> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"XNIO-1 I/O-1" prio=10 tid=0x00007fd9cc72d800 nid=0xcb3 runnable [0x00007fd98e3d1000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
	at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:269)
	at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:79)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
	- locked <0x00000000e0cd8780> (a sun.nio.ch.Util$2)
	- locked <0x00000000e0cd8770> (a java.util.Collections$UnmodifiableSet)
	- locked <0x00000000e0cd8528> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
	at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:502)

   Locked ownable synchronizers:
	- None

"Thread-3" prio=10 tid=0x00007fd9cc63a000 nid=0xcaa waiting on condition [0x00007fd98e6db000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at java.lang.Thread.sleep(Native Method)
	at org.syncany.gui.tray.TrayIcon$5.run(TrayIcon.java:469)
	at java.lang.Thread.run(Thread.java:745)

   Locked ownable synchronizers:
	- None

"Timer-0" prio=10 tid=0x00007fd9cc638800 nid=0xca9 in Object.wait() [0x00007fd994120000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.util.TimerThread.mainLoop(Timer.java:552)
	- locked <0x00000000e0cc0300> (a java.util.TaskQueue)
	at java.util.TimerThread.run(Timer.java:505)

   Locked ownable synchronizers:
	- None

"Service Thread" daemon prio=10 tid=0x00007fd9cc0af800 nid=0xc77 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread1" daemon prio=10 tid=0x00007fd9cc0ad000 nid=0xc76 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread0" daemon prio=10 tid=0x00007fd9cc0aa000 nid=0xc75 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Signal Dispatcher" daemon prio=10 tid=0x00007fd9cc0a8000 nid=0xc74 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Finalizer" daemon prio=10 tid=0x00007fd9cc07d000 nid=0xc71 in Object.wait() [0x00007fd9b63c7000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:135)
	- locked <0x00000000e0310a28> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:151)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)

   Locked ownable synchronizers:
	- None

"Reference Handler" daemon prio=10 tid=0x00007fd9cc07a800 nid=0xc70 in Object.wait() [0x00007fd9b64c8000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:503)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:133)
	- locked <0x00000000e0310bf0> (a java.lang.ref.Reference$Lock)

   Locked ownable synchronizers:
	- None

"main" prio=10 tid=0x00007fd9cc00b000 nid=0xc68 runnable [0x00007fd9d3a21000]
   java.lang.Thread.State: RUNNABLE
	at org.eclipse.swt.internal.gtk.OS.Call(Native Method)
	at org.eclipse.swt.widgets.Display.sleep(Unknown Source)
	at org.syncany.operations.gui.GuiOperation.startEventDispatchLoop(GuiOperation.java:207)
	at org.syncany.operations.gui.GuiOperation.execute(GuiOperation.java:99)
	at org.syncany.cli.GuiCommand.execute(GuiCommand.java:48)
	at org.syncany.cli.CommandLineClient.runLocally(CommandLineClient.java:389)
	at org.syncany.cli.CommandLineClient.runCommand(CommandLineClient.java:379)
	at org.syncany.cli.CommandLineClient.start(CommandLineClient.java:195)
	at org.syncany.Syncany.main(Syncany.java:55)

   Locked ownable synchronizers:
	- None

"VM Thread" prio=10 tid=0x00007fd9cc076800 nid=0xc6f runnable 

"GC task thread#0 (ParallelGC)" prio=10 tid=0x00007fd9cc020800 nid=0xc69 runnable 

"GC task thread#1 (ParallelGC)" prio=10 tid=0x00007fd9cc022800 nid=0xc6a runnable 

"GC task thread#2 (ParallelGC)" prio=10 tid=0x00007fd9cc024800 nid=0xc6b runnable 

"GC task thread#3 (ParallelGC)" prio=10 tid=0x00007fd9cc026800 nid=0xc6c runnable 

"VM Periodic Task Thread" prio=10 tid=0x00007fd9cc0ba000 nid=0xc78 waiting on condition 

JNI global references: 513
```

**Comment by pimotte** [comment]
I've been working on making Up less memory intensive: https://github.com/syncany/syncany/tree/bugfix/up-memory-usage-good-enough

As the name of the branch suggests, I've tried a ton of different things here. The problem in the end seems to be that the in memory caches I've replaced with database queries are actually a fairly significant slowdown when 10000+ files are stored. Normally two queries are done to guess previous history, and these take at least 50ms each, which means that 10 files per second is the limit.

There are a couple of questions I have:
1. Might this be acceptable? I'm feeling not, since I'm testing with a 300K file client and this is taking ages.
2. How could we resolve this? I tried swapping out HSQLDB for H2, which supports hash-based indexing. This give a decent speedup, enough to make it acceptable, but causes a bunch of problems in the rest of the code-base (ie. it only supports single-column subqueries).

**Comment by pimotte** [comment]
Just edited the SQL queries. It turns out that the views we have are fairly performance intensive. Looking much better now.

**Comment by pimotte** [comment]
Started working on Down memory usage at https://github.com/syncany/syncany/tree/bugfix/memory-usage-down.

@binwiederhier, is there any reason we want to retrieve the old fileversions (before down) from the database instead of from the filesystem? The whole system would be tremendously more simple if we can just apply the databaseversions, keep track of files that change, compare the filesystem versions to the ones in the database and make the changes.

**Comment by pimotte** [comment]
I have determined the reason to be: we cannot do this since we would just throw away any local changes.

**Comment by binwiederhier** [comment]
Yes, the system is designed to do a three-way comparison between remote, local and local database.

**Comment by binwiederhier** [comment]
With the release today, we have significantly reduced the memory footprint. I still don't consider 450 MB high for the work that we're doing. I'm closing this for now.
